From c44e4b4e51c67504d335464dd74bf5f530d25412 Mon Sep 17 00:00:00 2001
From: Shubhrajyoti Datta <shubhrajyoti.datta@xilinx.com>
Date: Sat, 3 Oct 2020 12:36:39 +0530
Subject: [PATCH 1700/1852] i2c-cadence: Check the return value of pin-ctrl

commit 02e5c803a5af1e85d75812623b8953e12902d6b8 from
https://github.com/Xilinx/linux-xlnx.git

Add condition to check the return value of pinctrl_select_okay.

Addresses-Coverity: Event check_return.
Signed-off-by: Shubhrajyoti Datta <shubhrajyoti.datta@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/i2c/busses/i2c-cadence.c | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/drivers/i2c/busses/i2c-cadence.c b/drivers/i2c/busses/i2c-cadence.c
index d6d8f055b9f0..7cd931f85ecb 100644
--- a/drivers/i2c/busses/i2c-cadence.c
+++ b/drivers/i2c/busses/i2c-cadence.c
@@ -1203,12 +1203,16 @@ static int __maybe_unused cdns_i2c_runtime_resume(struct device *dev)
 static void cdns_i2c_prepare_recovery(struct i2c_adapter *adapter)
 {
 	struct cdns_i2c *p_cdns_i2c;
+	int ret;
 
 	p_cdns_i2c = container_of(adapter, struct cdns_i2c, adap);
 
 	/* Setting pin state as gpio */
-	pinctrl_select_state(p_cdns_i2c->pinctrl,
+	ret = pinctrl_select_state(p_cdns_i2c->pinctrl,
 			     p_cdns_i2c->pinctrl_pins_gpio);
+	if (ret < 0)
+		dev_err(p_cdns_i2c->adap.dev.parent,
+				"pinctrl_select_state failed\n");
 }
 
 /**
@@ -1221,12 +1225,16 @@ static void cdns_i2c_prepare_recovery(struct i2c_adapter *adapter)
 static void cdns_i2c_unprepare_recovery(struct i2c_adapter *adapter)
 {
 	struct cdns_i2c *p_cdns_i2c;
+	int ret;
 
 	p_cdns_i2c = container_of(adapter, struct cdns_i2c, adap);
 
 	/* Setting pin state to default(i2c) */
-	pinctrl_select_state(p_cdns_i2c->pinctrl,
+	ret = pinctrl_select_state(p_cdns_i2c->pinctrl,
 			     p_cdns_i2c->pinctrl_pins_default);
+	if (ret < 0)
+		dev_err(p_cdns_i2c->adap.dev.parent,
+				"pinctrl_select_state failed\n");
 }
 
 /**
-- 
2.31.1

